component/azstorage/datalake.go

```diff
@@ -424,28 +384,31 @@ func (dl *Datalake) GetAttr(name string) (attr *internal.ObjAttr, err error) {
 	attr = &internal.ObjAttr{
 		Path:   name,
 		Name:   filepath.Base(name),
-		Size:   prop.ContentLength(),
+		Size:   *prop.ContentLength,
 		Mode:   mode,
 		Mtime:  lastModified,
 		Atime:  lastModified,
 		Ctime:  lastModified,
 		Crtime: lastModified,
 		Flags:  internal.NewFileBitMap(),
 	}
-	parseProperties(attr, prop.XMsProperties())
-	if azbfs.PathResourceDirectory == azbfs.PathResourceType(prop.XMsResourceType()) {
-		attr.Flags = internal.NewDirBitMap()
-		attr.Mode = attr.Mode | os.ModeDir
-	}
+	parseMetadata(attr, prop.Metadata)
+
+	// TODO:: track2: Add a check for resource type after sdk release
+	// if *prop.ResourceType == "directory" {
+	// 	attr.Flags.Set(uint16(internal.NewDirBitMap())) /
```

Comment:
This should be same as earlier
```suggestion
	// 	attr.Flags = internal.NewDirBitMap()
```